src/TrendsOverview.svelte

```diff
@@ -115,18 +124,21 @@
     setParentRegionButton();
 
     if (selectedCountryMetadata) {
-      mapData = fetchRegionalTrendLines(selectedCountryMetadata);
+      let temp_mapData = fetchRegionalTrendLines(selectedCountryMetadata);
 
-      mapData.then((mapData) => {
-        createMap(mapData, selectedMapTrendId, regions, onMapSelection, selectedCountryMetadata);
+      temp_mapData.then((mD) => {
+        $mapData = mD;
+        console.log(`mapData length is: ${$mapData.length}`)
+        createMap(selectedMapTrendId, regions, onMapSelection, selectedCountryMetadata);
         isMapInitialized = true;
         if (selectedRegion) {
           setMapSelection(selectedRegion);
         }
       });
+      let regionalTrendsData_temp = await fetchRegionalTrendsData(temp_mapData);
+      $regionalTrends = regionalTrendsData_temp;
```

Comment:
why use a temp variable instead of assigning to this directly?

Reply:
we are awaiting the result of a promise. I've renamed it to be more clear. is this better?

Reply:
yes. I can load it directly. I was worried initially that it would cause errors loading an empty map but seems ok.